src/hotspot/share/gc/g1/g1RemSet.cpp

```diff
@@ -841,7 +821,8 @@ class G1ScanCollectionSetRegionClosure : public HeapRegionClosure {
       event.commit(GCId::current(), _worker_id, G1GCPhaseTimes::phase_name(_scan_phase));
     }
 
-    if (_scan_state->claim_collection_set_region(region_idx)) {
+    // Scan code root remembered sets.
+    {
```

Comment:
Without the claim-logic, all workers will scan code-root. Why is it needed that multiples workers scan the same set of code-root repeatedly? I thought once is enough per region.

Reply:
Every hashtable has its internal claim now. So to have (eventually) multiple threads help with a single hashtable, all threads will at least to visit and check the current claim value whether this particular hashtable has been fully claimed (processed).